### PR TITLE
Support secret parameters for service instances

### PR DESCRIPTION
**Pull request by spadgett**
Use a secret for parameters when provisioning a service.

https://trello.com/c/IlDIJAz2

### CODE REVIEW

**Comment by spadgett** [comment]
@jeff-phillips-18 PTAL, this should work now with the latest catalog image

File: src/components/order-service/order-service.controller.ts [hunk 1]
```diff
@@ -218,18 +220,29 @@ export class OrderServiceController implements angular.IController {
...
-      this.ctrl.serviceInstance = data;
+      this.watchResults(resource, serviceInstance, context);
+      this.ctrl.serviceInstance = serviceInstance;
+
+      // Create the parameters secret if necessary.
+      if (secretName) {
+        let secret = this.makeParametersSecret(secretName, parameters, serviceInstance);
+        this.DataService.create('secrets', null, secret, context).then(_.noop, (e: any) => {
+          this.ctrl.error = _.get(e, 'data');
+        });
```
Comment (jeff-phillips-18) [inline src/components/order-service/order-service.controller.ts]: Is there an href to the secret that would get could get and pass to the bind results component now?

Comment (spadgett) [reply src/components/order-service/order-service.controller.ts]: There are two secrets now: one for the parameters and one for the bind result. I think the bind result is the interesting one. I'm not sure it's worth linking to the parameters secret.

Ideally we'd just have the instance/bind parameters on the provisioned service detail page.

Comment (jeff-phillips-18) [reply src/components/order-service/order-service.controller.ts]: OK, gotcha.

**Comment by spadgett** [comment]
Thanks @jeff-phillips-18

**Pull request merged**